### PR TITLE
[SPARK-26559][ML][PySpark] ML image can't work with numpy versions prior to 1.9

### PR DESCRIPTION
**Pull request by viirya**
## What changes were proposed in this pull request?

Due to [API change](https://github.com/numpy/numpy/pull/4257/files#diff-c39521d89f7e61d6c0c445d93b62f7dc) at 1.9, PySpark image doesn't work with numpy version prior to 1.9.

When running image test with numpy version prior to 1.9, we can see error:
```
test_read_images (pyspark.ml.tests.test_image.ImageReaderTest) ... ERROR                                                                              
test_read_images_multiple_times (pyspark.ml.tests.test_image.ImageReaderTest2) ... ok                                                                 
                                                                                                                                                      
======================================================================                                                                                
ERROR: test_read_images (pyspark.ml.tests.test_image.ImageReaderTest)                                                                                 
----------------------------------------------------------------------                                                                                
Traceback (most recent call last):
  File "/Users/viirya/docker_tmp/repos/spark-1/python/pyspark/ml/tests/test_image.py", line 36, in test_read_images                                   
    self.assertEqual(ImageSchema.toImage(array, origin=first_row[0]), first_row)                                                                      
  File "/Users/viirya/docker_tmp/repos/spark-1/python/pyspark/ml/image.py", line 193, in toImage                                                      
    data = bytearray(array.astype(dtype=np.uint8).ravel().tobytes())                                                                                  
AttributeError: 'numpy.ndarray' object has no attribute 'tobytes'                                                                                     
                                                                                                                                                      
----------------------------------------------------------------------
Ran 2 tests in 29.040s                                                                                                                                
                                                                                                                                                      
FAILED (errors=1)
```


## How was this patch tested?

Manually test with numpy version prior and after 1.9.

### CODE REVIEW

**Comment by viirya** [comment]
cc @HyukjinKwon

File: python/pyspark/ml/image.py [hunk 1]
```diff
@@ -190,7 +190,13 @@ def toImage(self, array, origin=""):
         # Running `bytearray(numpy.array([1]))` fails in specific Python versions
         # with a specific Numpy version, for example in Python 3.6.0 and NumPy 1.13.3.
         # Here, it avoids it by converting it to bytes.
-        data = bytearray(array.astype(dtype=np.uint8).ravel().tobytes())
+        numpy_version = np.__version__.split('.')
+        numpy_version_major = int(numpy_version[0])
+        numpy_version_minor = int(numpy_version[1])
+        if numpy_version_major > 1 or (numpy_version_major == 1 and numpy_version_minor >= 9):
```
Comment (HyukjinKwon) [inline python/pyspark/ml/image.py]: @viirya, not a big deal but I think using `LooseVersion` would be preferred

**Review by HyukjinKwon — APPROVED** [review]
Looks good if the tests pass

File: python/pyspark/ml/image.py (hunk 1 above)
Comment (viirya) [reply python/pyspark/ml/image.py]: Thanks for review. Changed to `LooseVersion`.

File: python/pyspark/ml/image.py [hunk 2]
```diff
@@ -190,7 +190,13 @@ def toImage(self, array, origin=""):
...
+            data = bytearray(array.astype(dtype=np.uint8).ravel().tobytes())
+        else:
+            data = bytearray(array.astype(dtype=np.uint8).ravel().tostring())
```
Comment (HyukjinKwon) [inline python/pyspark/ml/image.py]: Ah, actually, also @viirya, I think we can use `data = bytearray(array.astype(dtype=np.uint8).ravel())` right away. I only turned this into bytes for Python 3.6.0 and NumPy 1.13.3 - it has to support memoryview for perf.

Comment (viirya) [reply python/pyspark/ml/image.py]: oh, i see.

**Comment by HyukjinKwon** [comment]
LGTM

**Comment by viirya** [comment]
Thanks @HyukjinKwon

**Comment by HyukjinKwon** [comment]
Merged to master and branch-2.4.

**Pull request closed without merging**